src/OptimalFilter.jl

```diff
@@ -139,59 +139,64 @@ end
 
 # Normalized two-dimension discrete Fourier transofrm normalized by sqrt(n1_bar).
 # Operates on the 2d data stored as a matrix.
+# The argument `f` lets you switch between forward transform (`f=identity`) and
+# inverse transform (`f=inv`).
 # From Dietrich & Newsam 96 in text following equation 12
-function normalized_2d_fft!(transformed_array::AbstractMatrix{T}, array::AbstractMatrix{S}, grid_ext) where {T,S}
+function normalized_2d_fft!(transformed_array::AbstractMatrix{<:Complex}, array::AbstractMatrix{S}, fft_plan::FFTW.FFTWPlan, fft_plan!::FFTW.FFTWPlan, grid_ext, f=identity) where {S}
 
-    normalization_factor = 1.0 / sqrt(grid_ext.nx * grid_ext.ny)
-    transformed_array .= fft(array) .* normalization_factor
+    normalization_factor = f(sqrt(grid_ext.nx * grid_ext.ny))
+    if pointer(transformed_array) == pointer(array)
+        mul!(transformed_array, f(fft_plan!), array)
+    else
+        mul!(transformed_array, f(fft_plan), array)
```

Comment:
What's the difference between `fft_plan` and `fft_plan!`?

Reply:
The out-of-place `fft_plan` can be used only in
```julia
mul!(B, fft_plan, A)
```
with `A` and `B` different arrays, or more precisely `pointer(A) != pointer(B)`: https://github.com/JuliaMath/FFTW.jl/blob/2c20abe7098a2380fa37fe021b3d21fc593059ec/src/fft.jl#L438.  Instead, the in-place `fft_plan!` can only be used in
```julia
mul!(A, fft_plan!, A)
```
We do have cases where we overwrite the input array.